hedera-node/src/main/java/com/hedera/services/context/NodeInfo.java

```diff
@@ -61,14 +73,74 @@ public boolean isZeroStake(long nodeId) {
 		return isZeroStake[index];
 	}
 
-	private void readBook() {
+	/**
+	 * Convenience method to check if this node is zero-stake.
+	 */
+	public boolean isSelfZeroStake() {
+		return isZeroStake(selfId);
+	}
+
+	/**
+	 * Returns the account parsed from the address book memo corresponding
+	 * to the given node id.
+	 *
+	 * @param nodeId the id of interest
+	 * @throws IllegalArgumentException if the book did not contain the id, or was missing an account for the id
+	 */
+	public AccountID accountOf(long nodeId) {
+		if (!bookIsRead) {
+			readBook();
+		}
+
+		final int index = (int)nodeId;
+		if (index < 0 || index >= numberOfNodes) {
+			throw new IllegalArgumentException("No node with id " + nodeId + " was in the address book!");
+		}
```

Comment:
Duplicate code with what is in `isZeroStake(long nodeId)` except for the exception error message. Should extract it out with customized message.

Reply:
Extracted the repeated `if` condition into a `isIndexOutOfBounds()` test. 👍